test/spec/modules/pubCommonId_spec.js

```diff
@@ -0,0 +1,188 @@
+      let innerAdUnits1;
+      let innerAdUnits2;
+      let pubcid1;
+      let pubcid2;
+
+      requestBidHook({adUnits: adUnits1}, (config) => { innerAdUnits1 = config.adUnits });
+      pubcid1 = getCookie(COOKIE_NAME); // get first cookie
+      setCookie(COOKIE_NAME, '', -1); // erase cookie
+
+      innerAdUnits1.forEach((unit) => {
+        unit.bids.forEach((bid) => {
+          expect(bid).to.have.deep.property('crumbs.pubcid');
+          expect(bid.crumbs.pubcid).to.equal(pubcid1);
+        });
+      });
+
+      requestBidHook({adUnits: adUnits2}, (config) => { innerAdUnits2 = config.adUnits });
+      pubcid2 = getCookie(COOKIE_NAME); // get second cookie
+
+      innerAdUnits2.forEach((unit) => {
+        unit.bids.forEach((bid) => {
+          expect(bid).to.have.deep.property('crumbs.pubcid');
+          expect(bid.crumbs.pubcid).to.equal(pubcid2);
+        });
+      });
+
+      expect(pubcid1).to.not.equal(pubcid2);
+    });
+
+    it('Check new cookie', function () {
+      let adUnits = getAdUnits();
+      let innerAdUnits;
+      let pubcid = utils.generateUUID();
+
+      setCookie(COOKIE_NAME, pubcid, 600);
+      requestBidHook({adUnits}, (config) => { innerAdUnits = config.adUnits });
+      innerAdUnits.forEach((unit) => {
+        unit.bids.forEach((bid) => {
+          expect(bid).to.have.deep.property('crumbs.pubcid');
+          expect(bid.crumbs.pubcid).to.equal(pubcid);
+        });
+      });
+    });
+  });
+
+  describe('Configuration', function () {
+    it('empty config', function () {
+      // this should work as usual
+      setConfig({});
+      let adUnits = getAdUnits();
+      let innerAdUnits;
+      requestBidHook({adUnits}, (config) => { innerAdUnits = config.adUnits });
+      let pubcid = getCookie(COOKIE_NAME);
+      innerAdUnits.forEach((unit) => {
+        unit.bids.forEach((bid) => {
+          expect(bid).to.have.deep.property('crumbs.pubcid');
+          expect(bid.crumbs.pubcid).to.equal(pubcid);
+        });
+      });
+    });
+
+    it('disable', function () {
+      setConfig({enable: false});
+      setCookie(COOKIE_NAME, '', -1); // erase cookie
+      let adUnits = getAdUnits();
+      let unmodified = getAdUnits();
+      let innerAdUnits;
+      expect(isPubcidEnabled()).to.be.false;
+      requestBidHook({adUnits}, (config) => { innerAdUnits = config.adUnits });
+      expect(getCookie(COOKIE_NAME)).to.be.null;
+      assert.deepEqual(innerAdUnits, unmodified);
+      setConfig({enable: true}); // reset
+      requestBidHook({adUnits}, (config) => { innerAdUnits = config.adUnits });
+      innerAdUnits.forEach((unit) => {
+        unit.bids.forEach((bid) => {
+          expect(bid).to.have.deep.property('crumbs.pubcid');
+        });
+      });
+    });
+
+    it('change expiration time', function () {
+      setConfig({expInterval: 100});
+      setCookie(COOKIE_NAME, '', -1); // erase cookie
+      expect(getExpInterval()).to.equal(100);
```

Comment:
Is everything below this needed to validate that the cookie expiration interval has changed?

Reply:
Yeah, it was just to make sure the cookie can still be retrieved after the interval change.  Not strictly needed.